pkg/cli/config.go

```diff
@@ -79,3 +109,84 @@ func expandPath(path string) string {
 	}
 	return strings.ReplaceAll(path, "~", home)
 }
+
+// Build a final configuration instance from different sources. Each config parameter will take
+// the value from the first source where it's set.
+func buildConfig(sources []io.Reader) (*Config, error) {
```

Comment:
Would this work instead?
```
func buildConfig(sources []io.Reader) (*Config, error) {
  c := nonZeroDefaults()
  for _, s := range sources {
    if err := toml.NewDecoder(s).Decode(c); err != nil {
      return err
    }
  }
  return c, nil
}
func nonZeroDefaults() *Config {...}
```

It seems to work for JSON: https://goplay.tools/snippet/IVV86tVxlIB

Reply:
worked as a charm! great catch, thanks.